packages/react-components/src/components/timegraph-output-component.tsx

```diff
@@ -504,9 +518,49 @@ export class TimegraphOutputComponent extends AbstractTreeOutputComponent<Timegr
         return <div id='main-timegraph-content' ref={this.horizontalContainer} style={{ height: this.props.style.height }} >
             {this.getChartContainer()}
             {this.getMarkersContainer()}
+            <div id={this.props.traceId + this.props.outputDescriptor.id + 'searchBar'} className='timgraph-search-bar'>
+                <TextField
+                    InputProps={{
+                        placeholder: "Search",
+                        startAdornment: (
+                            <InputAdornment
+                                sx={{
+                                    color: "var(--theia-ui-font-color0)"
+                                }}
+                                position="start"
+                            >
+                                <i className='codicon codicon-search'></i>
+                            </InputAdornment>
+                        ),
+                        className: "timegraph-search-box",
+                        endAdornment: (
+                            <InputAdornment 
+                                sx={{
+                                    color: "var(--theia-ui-font-color0)"
+                                }}
+                                position="end"
+                            >
+                                <button className='remove-search-button' onClick={this.clearSearchBox}>
+                                    <i className='codicon codicon-close'></i>
+                                </button>
+                            </InputAdornment>
+                        )
+                    }}
+                    value={this.state.searchString}
+                    onChange={this.handleSearchChange}
+                />
+            </div>
         </div>;
     }
 
+    private handleSearchChange(event: React.ChangeEvent<HTMLInputElement>) {
+        this.setState({ searchString: event.target.value ?? '' });
```

Comment:
I noticed that when typing an equal sign "=", the timeline-chart zooms out, because it's a keyboard shortcut. So, when the focus is on the textbox the key events must not be handled in the timeline-chart.